.gitlab-ci.yml

```diff
@@ -18,7 +18,7 @@ pre-commit:
 tox:
   parallel:
     matrix:
-      - BUILD_PY_VERSION: [python3.7, python3.8, python3.9, python3.10, python3.11]
+      - BUILD_PY_VERSION: [python3.7, python3.8, python3.9, python3.10]
```

Comment:
I noticed the tox.ini only includes py38. Should we update that to include the additional versions here that are defined for gitlab?

Reply:
I think it might be a good idea to keep it to one python version in tox. This matrix section should override the env section in tox. Otherwise the default `tox` run locally will run the test suite 4 times.

Reply:
This may have been a bad recommend on my part. My motivation was to keep the files in sync so that if a developer wanted to run tests against all the python versions that gitlab was going to run, that would be an easy option. Otherwise, they could use the -e flag to specify a particular environment.

It seems like there are other concerns here. Maybe we should enact a policy that the tox.ini only has a single target python version, regardless of whether it's a service or library (since gitlab will catch issues with the full set of python versions) and the tox.ini can specify the preferred version of python to use? @ProfOak and @kulgan , thoughts on that?

And that brings up another question. If that's is our intention, what is the appropriate version of python to specify in the tox.ini? Why would we default to python 3.8 instead of either the min or max version?